nncf/experimental/openvino_native/graph/model_transformer.py

```diff
@@ -87,41 +89,48 @@ def transform(self, transformation_layout: TransformationLayout) -> ov.Model:
                 model_extraction_transformation = transformation
             elif isinstance(transformation, OVBiasCorrectionCommand):
                 bias_correction_transformations.append(transformation)
-
-        if output_insertion_transformations:
-            self._apply_output_insertion_transformations(output_insertion_transformations)
+        model = None
+        # Inplace transformations; Using deepcopy of model
         if fq_nodes_removing_transformations:
-            self._apply_fq_nodes_removing_transformation(fq_nodes_removing_transformations)
+            model = self._apply_fq_nodes_removing_transformation(fq_nodes_removing_transformations)
         if quantizer_insertion_transformations:
-            self._apply_quantizer_insertion_transformations(quantizer_insertion_transformations)
+            model = self._apply_quantizer_insertion_transformations(quantizer_insertion_transformations)
         if bias_correction_transformations:
-            self._apply_bias_correction_transformations(bias_correction_transformations)
+            model = self._apply_bias_correction_transformations(bias_correction_transformations)
         if model_extraction_transformation:
-            self._model = self._apply_model_extraction_transformation(model_extraction_transformation)
-
-        return self._model
-
-    def _apply_output_insertion_transformations(self, transformations: List[OVOutputInsertionCommand]) -> None:
+            model = self._apply_model_extraction_transformation(model_extraction_transformation)
+        if output_insertion_transformations:
+            model = self._apply_output_insertion_transformations(output_insertion_transformations)
+        # No transformation applied
+        if model is None:
+            logger.warning('No transformations were applied to the model. The copy of the model is returned.')
+            return self._model.clone()
+        return model
+
+    def _apply_output_insertion_transformations(self,
+                                                transformations: List[OVOutputInsertionCommand]) -> ov.Model:
         """
         Applies incoming transformations to the model.
 
         :param transformations: list of the OVOutputInsertionCommand transformations.
         """
-        extra_model_outputs = self._get_extra_model_outputs(transformations)
-        self._model = self._insert_outputs(self._model, outputs=extra_model_outputs)
+        model = self._model.clone()
```

Comment:
Another question: is it valid to make a copy while applying every transformation type? I mean, in the case of the transformations by transformations pipeline (for example, add outputs and quantizer), we will make an unnecessary model copy on the second stage (inserting adding quantizer). If copies will be made at every stage, it will break the essence of the ModelTransformer.